lua/nvim-tree/renderer/decorator/hidden.lua

```diff
@@ -0,0 +1,60 @@
+local is_dotfile = require("nvim-tree.explorer.node").is_dotfile
+local HL_POSITION = require("nvim-tree.enum").HL_POSITION
+local ICON_PLACEMENT = require("nvim-tree.enum").ICON_PLACEMENT
+
+local Decorator = require "nvim-tree.renderer.decorator"
+
+---@class DecoratorHidden: Decorator
+---@field icon HighlightedString|nil
+local DecoratorHidden = Decorator:new()
+
+---@param opts table
+---@return DecoratorHidden
+function DecoratorHidden:new(opts)
+  local o = Decorator.new(self, {
+    enabled = true,
+    hl_pos = HL_POSITION[opts.renderer.highlight_hidden] or HL_POSITION.none,
+    icon_placement = ICON_PLACEMENT[opts.renderer.icons.hidden_placement] or ICON_PLACEMENT.none,
+  })
+  ---@cast o DecoratorHidden
+
+  if not o.enabled then
+    return o
+  end
+
+  if opts.renderer.icons.show.hidden then
+    o.icon = {
+      str = opts.renderer.icons.glyphs.hidden,
+      hl = { "NvimTreeHiddenIcon" },
+    }
+    o:define_sign(o.icon)
+  end
+
+  return o
+end
+
+---Hidden icon: hidden.enable, renderer.icons.show.hidden and node starts with `.` (dotfile).
+---@param node Node
+---@return HighlightedString[]|nil icons
+function DecoratorHidden:calculate_icons(node)
+  if self.enabled and is_dotfile(node) then
```

Comment:
```suggestion
  if self.enabled and node:is_dotfile() then
```

You can then remove the require.

Reply:
I've updated in favor of all sugestions. However, I couldn't use ``is_dotfile`` as a method of ``node``. 
Instead I followed the same style of other decorators by 'requiring' into a variable called ``explorer_node`` and then using ``explorer_node.is_dotfile(node)``. 
Besides that, everything else should be aligned with the change request.

Reply:
> However, I couldn't use `is_dotfile` as a method of `node`. Instead I followed the same style of other decorators by 'requiring' into a variable called `explorer_node` and then using `explorer_node.is_dotfile(node)`.

Right you are; node isn't yet a proper class. Apologies...